### ISSUE TITLE
contentText for before & after decorations does not always appear

### ISSUE THREAD

**Issue by siegebell** [Microsoft/vscode#8003]
- VSCode Version: 1.3.0-insider; commit  a6aa320 
- OS Version: Windows 8

Issues:
1. empty-range after-content-text does not appear at the beginning of lines
2. empty-range before-content-text does not appear at the end of lines
3. neither before- nor after-content can appear on empty lines

see: #7181

MWE:
```typescript
  const decBefore = vscode.window.createTextEditorDecorationType({
    before: { contentText: "$BEFORE-CONTENT$" }
  });
  const decAfter = vscode.window.createTextEditorDecorationType({
    after: { contentText: "$AFTER-CONTENT$" }
  });
  const editor = vscode.window.activeTextEditor;
  if (editor) {
    const lines = [
      "after-content at the beginning of this line (does not appear; bug)",
      "after-content at end of this line (ok)",
      "after-content on an empty line (does not appear; bug):",
      "",
      "before-content at beginning of this line (ok)",
      "before-content at end of this line (does not appear; bug)",
      "before-content on an empty line (does not appear; bug):",
      "", "END OF EXAMPLE", "" ]
    editor.edit((edit) => {
      edit.insert(new vscode.Position(0,0), lines.join('\n'));
    }).then(() => {
      // Note the empty ranges
      // after-content appears at the second range but not the first or third
      editor.setDecorations(decAfter, [new vscode.Range(0,0,0,0), new vscode.Range(1,lines[1].length,1,lines[1].length), new vscode.Range(3,0,3,0)]);
      // before-content appears at the first range but not the second or third
      editor.setDecorations(decBefore, [new vscode.Range(4,0,4,0), new vscode.Range(5,lines[5].length,5,lines[5].length), new vscode.Range(7,0,7,0)]);
    });
  }
```

**Comment by siegebell** [comment]
Observed:
![before-after-bug](https://cloud.githubusercontent.com/assets/16118166/16273139/83b51492-386e-11e6-827c-ab78816c1c4e.PNG)

Expected:
![before-after-expected](https://cloud.githubusercontent.com/assets/16118166/16273140/83c6ca3e-386e-11e6-9f53-f76e6a4b5fa1.PNG)

**Comment by dbaeumer** [comment]
@aeschli FYI

**Comment by aeschli** [comment]
@siegebell Thanks a lot for the great example!

The problem is that we need a range of length >= 1. That unfortunately prevents annotating empty lines.
We will have to think about it more; there's no easy fix.

Closing as duplicate of #8341.
